cirq/ops/pauli_string.py

```diff
@@ -1008,63 +1054,292 @@ def _from_json_dict_(  # type: ignore
         return cls(pauli=pauli, qubit=qubit)
 
 
-class _MutablePauliString:
+class MutablePauliString:
+
+    def __init__(self,
+                 *contents: 'cirq.PAULI_STRING_LIKE',
+                 coefficient: Union[int, float, complex] = 1,
+                 pauli_int_dict: Optional[Dict['cirq.Qid', int]] = None):
+        self.coefficient = complex(coefficient)
+        self.pauli_int_dict = {} if pauli_int_dict is None else pauli_int_dict
+        if contents:
+            self.inplace_left_multiply_by(contents)
+
+    def _imul_atom_helper(self, qubit: 'cirq.Qid', pauli_lhs: int,
+                          sign: int) -> int:
+        pauli_old = self.pauli_int_dict.pop(qubit, 0)
+        pauli_new = pauli_lhs ^ pauli_old
+        if pauli_new:
+            self.pauli_int_dict[qubit] = pauli_new
```

Comment:
I'm getting the value using `dict.pop` instead of `dict.get`, which removes it from the dictionary as part of getting it. It's my latest favorite method that I somehow didn't know about.

Reply:
Cool :-)